src/shell/shell_cmds.cpp

```diff
@@ -1390,7 +1455,15 @@ void DOS_Shell::CMD_DATE(char * args) {
 	}
 	// check if a date was passed in command line
 	uint32_t newday, newmonth, newyear;
-	if (sscanf(args, "%u-%u-%u", &newmonth, &newday, &newyear) == 3) {
+	int n = dos.tables.country[0] == 1
+	                ? sscanf(args, "%u%c%u%c%u", &newday, &c1, &newmonth,
```

Comment:
After sscanf has convert the user's string into the raw numbers - it seems this PR should have some common check functions: `bool is_date_valid(y, m, d)`, and `is_time_valid(h, m, s)`.
(unless it exists somewhere, but I'm not seeing it?)
Maybe C++'s <chrono> has some time and data parsing-from-string routines too.

Reply:
Done.